### PR TITLE
Dynamic Diagram Revamp

### PR DESCRIPTION
**Pull request by fdodino**
- separating two colors for custom/wre objects
- removing root references from named objects
- short descriptions for Pair, Range and Position
- List, Set, Dictionary are shown with an ellipse
- Colorblind feature (special colors & 2 line border for objects)
- the user can select & remove objects from the diagram
- dynamic diagram toolbar shows how many hidden objects are and you can bring them back
- enhanced algorithm for showing objects (detecting largest width)
- showing transition effect when object references change
  - transition is optional (selectable from the toolbar)
- self references have a different anchor (you'll see it better)
- avoid entering in loop for cyclic references
- constant references appear with a lock icon

Link to the [specification](https://docs.google.com/document/d/1XaQ1rzMoRWyGudNJI73slwDi-a_3SgEk4OygQdFWkXk/edit#)

I also developed additional features:
- static diagram toolbar was revamped too (better icons)
- outline icons for variable, constant & properties were enhanced

### CODE REVIEW

File: org.uqbar.project.wollok.ui.diagrams/src/org/uqbar/project/wollok/ui/diagrams/dynamic/parts/VariableModel.xtend [hunk 1]
```diff
@@ -209,13 +217,25 @@ class VariableModel extends Shape {
...
-		this.bounds.top.y
+		location.y
 	}
 
 	def int getYValueForAnchor() {
 		YItShouldHave + (this.size.height / 2) - PADDING
 	}
 
+	override getBounds() {
+		if (DynamicDiagramConfiguration.instance.firstTimeRefreshView && !DynamicDiagramView.hadVariable(this)) {
```
Comment (PalumboN) [inline org.uqbar.project.wollok.ui.diagrams/src/org/uqbar/project/wollok/ui/diagrams/dynamic/parts/VariableModel.xtend:228]: `shouldAnimate()`?

File: org.uqbar.project.wollok/src/org/uqbar/project/wollok/interpreter/core/WollokNativeLobby.xtend [hunk 2]
```diff
@@ -22,6 +23,7 @@ import org.uqbar.project.wollok.interpreter.nativeobj.AbstractWollokDeclarativeN
  */
 class WollokNativeLobby extends AbstractWollokDeclarativeNativeObject implements EvaluationContext<WollokObject> {
 	val Map<String, WollokObject> localProgramVariables = newHashMap
+	List<String> constantVariables = newArrayList
```
Comment (PalumboN) [inline org.uqbar.project.wollok/src/org/uqbar/project/wollok/interpreter/core/WollokNativeLobby.xtend:26]: Por qué esto acá y no en el interpreter?

File: org.uqbar.project.wollok/src/org/uqbar/project/wollok/model/WollokModelExtensions.xtend [hunk 3]
```diff
@@ -109,6 +110,7 @@ class WollokModelExtensions {
 	def static dispatch String fqn(WNamedObject it) { nameWithPackage }
 	def static dispatch String fqn(WMixin it) { nameWithPackage }
 	def static dispatch String fqn(WSuite it) { nameWithPackage }
+	def static dispatch String fqn(MixedMethodContainer it) { nameWithPackage }
```
Comment (PalumboN) [inline org.uqbar.project.wollok/src/org/uqbar/project/wollok/model/WollokModelExtensions.xtend:113]: :laughing:

File: org.uqbar.project.wollok/src/org/uqbar/project/wollok/interpreter/core/WollokNativeLobby.xtend (hunk 2 above)
Comment (fdodino) [reply org.uqbar.project.wollok/src/org/uqbar/project/wollok/interpreter/core/WollokNativeLobby.xtend:26]: uh, ni se me pasó por la cabeza, fui por WollokObject y vi dónde más era necesario. Podríamos dejarlo para un refactor más adelante...

File: org.uqbar.project.wollok.ui.diagrams/src/org/uqbar/project/wollok/ui/diagrams/dynamic/parts/VariableModel.xtend (hunk 1 above)
Comment (fdodino) [reply org.uqbar.project.wollok.ui.diagrams/src/org/uqbar/project/wollok/ui/diagrams/dynamic/parts/VariableModel.xtend:228]: sí, `shouldAnimate()` es de más alto nivel, sin dudas... le meto ese cambio

**Pull request merged**